packages/arb-token-bridge-ui/src/components/TransferPanel/USDCDeposit/USDCDepositConfirmationDialog.tsx

```diff
@@ -0,0 +1,178 @@
+import { useState } from 'react'
+import { Tab, Dialog as HeadlessUIDialog } from '@headlessui/react'
+import { XMarkIcon } from '@heroicons/react/24/outline'
+
+import { Dialog, UseDialogProps } from '../../common/Dialog'
+import { Button } from '../../common/Button'
+import { ExternalLink } from '../../common/ExternalLink'
+import {
+  USDCBridgeInfo,
+  FastBridgeNames,
+  getFastBridges,
+  SpecialTokenSymbol
+} from '../../../util/fastBridges'
+import { TabButton } from '../../common/Tab'
+import { BridgesTable } from '../../common/BridgesTable'
+import { useAppState } from '../../../state'
+import { useNetworksAndSigners } from '../../../hooks/useNetworksAndSigners'
+import { getNetworkName, isNetwork } from '../../../util/networks'
+import { trackEvent } from '../../../util/AnalyticsUtils'
+import { useIsConnectedToArbitrum } from '../../../hooks/useIsConnectedToArbitrum'
+import { CommonAddress } from '../../../util/CommonAddressUtils'
+import { USDCDepositConfirmationDialogCheckbox } from './USDCDepositConfirmationDialogCheckbox'
+
+export function USDCDepositConfirmationDialog(
+  props: UseDialogProps & { amount: string }
+) {
+  const {
+    app: { selectedToken }
+  } = useAppState()
+  const { l1, l2 } = useNetworksAndSigners()
+  const isConnectedToArbitrum = useIsConnectedToArbitrum()
+  const networkName = getNetworkName(l2.network.id)
+  const { isArbitrumOne } = isNetwork(l2.network.id)
+
+  const [USDCcheckboxChecked, setUSDCcheckboxChecked] = useState(false)
+
+  const from = isConnectedToArbitrum ? l2.network : l1.network
+  const to = isConnectedToArbitrum ? l1.network : l2.network
+
+  const tokenSymbol = selectedToken?.symbol as SpecialTokenSymbol.USDC
+
+  const fastBridges = [
+    ...getFastBridges({
+      from: from.id,
+      to: to.id,
+      tokenSymbol,
+      fromTokenAddress: CommonAddress.Mainnet.USDC,
+      toTokenAddress: CommonAddress.ArbitrumOne.USDC,
+      amount: props.amount
+    })
+  ].filter(bridge => {
+    return (
+      tokenSymbol &&
+      (USDCBridgeInfo.supportedBridges as readonly FastBridgeNames[]).includes(
+        bridge.name
+      )
+    )
+  })
+
```

Comment:
All the checks for `tokenSymbol` seem unnecessary, it should never be the case that it's null, right? Can we early return `null` from the component in case it's `null`, just to make TS happy?